ccmlib/scylla_docker_cluster.py

```diff
@@ -24,30 +25,78 @@ class ScyllaDockerNode(ScyllaNode):
             dirs[i] = os.path.join(self.get_path(), i)
         return dirs
 
+    @staticmethod
+    def get_docker_name():
+        return run(["docker", "ps", "-a"], stdout=PIPE).stdout.decode('utf-8').split()[-1]
+
     def update_yaml(self):
-        pass # TODO: handle as mount point ?
+        pass  # TODO: handle as mount point ?
+
+    def create_docker(self):
+        res = run(['bash', '-c', f"docker run -v {self.local_yaml_path}:/etc/scylla/ --name {self.docker_name} -d {self.cluster.docker_image} --smp 1"], stdout=PIPE, stderr=PIPE)
+        self.docker_id = res.stdout.decode('utf-8').strip() if res.stdout else None
+        return res.returncode == 0, res.stdout, res.stderr
+
+    def service_start(self, service_name):
+        res = run(['docker', 'exec', self.docker_name, 'supervisorctl', 'start', service_name],
+                  stdout=PIPE, stderr=PIPE)
+        if res.returncode != 0:
+            print(res.stdout)
+            print(f'service {service_name} failed to start with error\n{res.stderr}')
+
+    def service_stop(self, service_name):
+        res = run(['docker', 'exec', self.docker_name, 'supervisorctl', 'stop', service_name],
+                  stdout=PIPE, stderr=PIPE)
+        if res.returncode != 0:
+            print(res.stdout)
+            print(f'service {service_name} failed to stop with error\n{res.stderr}')
+
+    def service_status(self, service_name):
+        res = run(['docker', 'exec', self.docker_name, 'supervisorctl', 'status', service_name],
+                  stdout=PIPE, stderr=PIPE)
+        if res.returncode != 0:
+            print(res.stdout)
+            print(f'service {service_name} failed to stop with error\n{res.stderr}')
+        else:
+            return res.stdout.decode('utf-8').split()[1]
 
     def start(self, *args, **kwargs):
         # TODO: find a better place to do this trick, at initializing maybe ? only if file doesn't exist in the test cluster dir yet ? as part as `update_yaml` ?
         # get scylla.yaml out of the docker
-        local_yaml_path = os.path.join(self.get_path(), 'conf', 'scylla.yaml')
-        run(['bash', '-c',  f'docker run --rm --entrypoint cat {self.cluster.docker_image}  /etc/scylla/scylla.yaml > {local_yaml_path}'])
+        run(['bash', '-c', f'docker run --rm --entrypoint cat {self.cluster.docker_image}  /etc/scylla/scylla.yaml > {self.local_yaml_path}/scylla.yaml'])
+        # res = run([f'docker run --rm --entrypoint cat {self.cluster.docker_image}  '
+        #            f'/etc/scylla/scylla.yaml > {self.local_yaml_path}'],
+        #           stdout=PIPE, stderr=PIPE)
+        # if not res.returncode:
+        #     print(f'Failed to copy scylla.yaml to {self.local_yaml_path}\n{res.stderr}')
+        rc, out, err = self.create_docker()
+        if not rc:
+            raise BaseException(f'failed to create docker {self.docker_name}')
         super(ScyllaDockerNode, self).start(*args, **kwargs)
+        return random.randint(100, 10000)
```

Comment:
Why are we returning a random number here ?
To simulate a pid number ?

I don't think I would be helpful, we will need to overwrite all this pid handling code somehow...

Reply:
yeah, the idea was to simulate a PID number, but it didn't work either (and i forgot to remove it)

and yes, we will need to make changes in all this PID handling (it might be easier than we think, if we override the methods in the base node code to use the docker classes)